Hardcore.lua

```diff
@@ -706,14 +729,35 @@ end
 function Hardcore:CHAT_MSG_ADDON(prefix, datastr, scope, sender)
 	-- Ignore messages that are not ours
 	if COMM_NAME == prefix then
-		-- Get the command
 		local command, data = string.split(COMM_COMMAND_DELIM, datastr)
-
 		-- Determine what command was sent
 		if command == COMM_COMMANDS[2] then
 			Hardcore:Add(data)
 		elseif command == COMM_COMMANDS[1] then
 			Hardcore:ReceivePulse(data, sender)
+		elseif command == COMM_COMMANDS[3] then -- send recover time ack
+			if CTL and isInGuild and guild_player_first_ping_time[sender] ~= nil and pulses[sender] ~= nil then
+				local commMessage = COMM_COMMANDS[4] .. COMM_COMMAND_DELIM .. tostring(guild_player_first_ping_time[sender]) .. COMM_COMMAND_DELIM .. tostring(pulses[sender])
+				CTL:SendAddonMessage("BULK", COMM_NAME, commMessage, "WHISPER", sender)
+			end
```

Comment:
To account for latency, I'd suggest generating a random milliseconds range from 1ms to 15000ms for complying with the request. If an eligible client is waiting for their time to send and another person sends earlier than them, they can cancel their pending send.

That will cut down on 100 guild members sending all at the same time